### ISSUE TITLE
Generated Stan code for NLE-MDN rebuilds every mixture Cholesky factor once per observation

### ISSUE THREAD

**Issue by pedroliman** [pedroliman/neuralsbi#313]
`stan_fn_mdn()` in `R/stan.R` generates `<name>_sum_lpdf()` for repeated i.i.d. observations under an NLE-MDN fit. `stan_sum_lines()` correctly hoists `head <- <name>_head(ts, w)` (the MLP forward pass over `theta`) outside the `for (n in 1:rows(x))` loop, since it depends only on `theta`. That mirrors `de_iid_evaluator.nsbi_de_mdn()` on the R side, which builds the mixture parameters once per `theta` and reuses them across observations.

`<name>_from_head()` (called once per row of `x` inside that loop) still reconstructs every mixture component's Cholesky factor `L` from `head`'s flat lower-triangular block on every call (lines ~498-503):

```
for (k in 1:K) {
  matrix[P, P] L = rep_matrix(0.0, P, P);
  vector[P] mu = mflat[...];
  { ...tril_lines: softplus-diagonal assignment into L... }
  lp[k] = logits[k] + multi_normal_cholesky_lpdf(xs | mu, L);
}
```

`head` (hence `mu`/`L`/`logits` for every component) is identical across all `N` observations, so this K-component Cholesky assembly runs `N` times instead of once per NLE evaluation. That's on every leapfrog step of every NUTS iteration for an NLE model fit with an MDN estimator and exported to Stan — exactly the "many i.i.d. observations" case `nle()` + Stan export exists for (per `CLAUDE.md`: "the log-likelihood of repeated i.i.d. observations is a sum over one trained density rather than a retrained network").

Fix direction: split `<name>_from_head()` so the per-component `mu`/`L` (and `logits`) are built once, in the `precompute` block passed to `stan_sum_lines()` alongside `head`, and only `multi_normal_cholesky_lpdf(xs | mu[k], L[k])` plus the mixture `log_sum_exp` runs inside the observation loop. Correctness is unaffected; this is purely a generated-code performance issue.

Relevant code: `R/stan.R` (`stan_fn_mdn`, `stan_sum_lines`), for comparison `R/likelihood.R` (`mdn_mixture`, `de_iid_evaluator.nsbi_de_mdn`) and `R/mdn.R` (`mdn_build_tril`).
